src/Altinn.AccessManagement.Persistence/ResourceMetadataRepo.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Data;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Altinn.AccessManagement.Core.Models;
+using Altinn.AccessManagement.Core.Repositories.Interfaces;
+using Altinn.AccessManagement.Persistence.Configuration;
+using Altinn.AccessManagement.Persistence.Extensions;
+using Dapper;
+using Npgsql;
+
+namespace Altinn.AccessManagement.Persistence
+{
+    /// <summary>
+    /// Storing Resource Rigistry metadata to Access management
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class ResourceMetadataRepo : IResourceMetadataRepository
+    {
+        private readonly IDbConnection _connection;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ResourceMetadataRepo"/> class
+        /// </summary>
+        /// <param name="dbConnection">Database connection for AuthorizationDb</param>
+        public ResourceMetadataRepo(NpgsqlDataSource dbConnection)
+        {
+            var bld = new NpgsqlConnectionStringBuilder(dbConnection.ConnectionString);
+            bld.AutoPrepareMinUsages = 2;
+            bld.MaxAutoPrepare = 50;
+            _connection = new Npgsql.NpgsqlConnection(bld.ConnectionString);
+        }
+
+        /// <inheritdoc />
+        public async Task<AccessManagementResource> InsertAccessManagementResource(AccessManagementResource resource, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+
+            var param = new Dictionary<string, object>();
+            param.Add("resourceregistryid", resource.ResourceRegistryId);
+            param.Add("resourcetype", resource.ResourceType.ToString().ToLower());
+
+            string query = $"INSERT INTO accessmanagement.resource (resourceregistryid, resourcetype, created, modified)" +
+                $"VALUES (@resourceregistryid, @resourcetype, now(), now())" +
+                $"ON CONFLICT (resourceregistryid) DO UPDATE SET resourcetype = @resourcetype, modified = now();" +
+                $"SELECT r.resourceid, r.resourceregistryid, r.resourcetype, r.created, r.modified" +
+                $" FROM accessmanagement.resource r" +
+                $" WHERE r.resourceregistryid = @resourceregistryid";
```

Comment:
Par tips: lengre spoerringer kan bruke raw literals saa slipper man string concatinations.
Also, man trenger ikke 2 statements da pgsql stoetter aa returnere inserted data.

```suggestion
            string query = 
                """
                INSERT INTO accessmanagement.resource (resourceregistryid, resourcetype, created, modified)
                VALUES (@resourceregistryid, @resourcetype, now(), now())
                ON CONFLICT (resourceregistryid) DO UPDATE SET resourcetype = @resourcetype, modified = now()
                RETURNING resourceid, resourceregistryid, resourcetype, created, modified;
                """;
```

Reply:
Also, syntax highlighting in visual studio:

![image](https://github.com/Altinn/altinn-access-management/assets/112334/8a74b199-2075-4a68-86b6-e2c50a71b144)

The comment, combined with a plugin to visual studio enables this.